### PR TITLE
Implement show for IncrementalCompact

### PR DESCRIPTION
**Pull request by Keno**
Basically works, but I think the formatting of the breaks need some slight improvement. @oscardssmith could you take this over?

### CODE REVIEW

**Comment by oscardssmith** [comment]
will do.

File: base/compiler/ssair/show.jl [hunk 1]
```diff
@@ -816,32 +832,17 @@ function show_ir(io::IO, compact::IncrementalCompact, config::IRShowConfig=defau
...
         printstyled("─"^width, color=:red)
     end
-    for idx in compact.idx:length(stmts)
-        if config.should_print_stmt(compact, idx, used)
-            bb_idx = show_ir_stmt(io, compact, idx, config, used, cfg, bb_idx; pop_new_node!)
-        elseif bb_idx <= length(cfg.blocks) && idx == cfg.blocks[bb_idx].stmts.stop
-            bb_idx += 1
-        end
-    end
+    show_ir_helper(io::IO, compact.idx:length(stmts), compact, config, used, cfg, bb_idx, pop_new_node!)
```
Comment (KristofferC) [inline base/compiler/ssair/show.jl:846]: Is the `::IO` needed?

Comment (oscardssmith) [reply base/compiler/ssair/show.jl:846]: oops. typo from copy-paste.

File: base/compiler/ssair/show.jl [hunk 2]
```diff
@@ -770,24 +789,58 @@ function default_config(code::IRCode; verbose_linetable=false)
...
+    used = BitSet(i for (i, x) in pairs(compact.used_ssas) if x != 0)
+
+    # First print everything that has already been compacted
+    pop_new_node! = ircode_new_nodes_iter(compact)
+    bb_idx = show_ir_helper(io, 1:compact.result_idx-1, compact, config, used, cfg, 1, pop_new_node!)
+    # Print unompacted nodes from the original ir
+    stmts = compact.ir.stmts
+    pop_new_node! = ircode_new_nodes_iter(compact.ir)
+    if compact.idx < length(stmts)
+        printstyled("─"^width, color=:red)
```
Comment (Keno) [inline base/compiler/ssair/show.jl:840]: This line is a little awkward right now - probably needs to be indented a bit to match the framing and be slightly shorter - just play with the formatting a bit.

Comment (oscardssmith) [reply base/compiler/ssair/show.jl]: how's this?

**Comment by oscardssmith** [comment]
is this good to merge?

**Comment by Keno** [comment]
I guess, this should probably have a test since as demonstrated this is easy to break. Can just be a simple printing for something.

**Comment by Keno** [comment]
I think there is a couple further thing that should be done here:
1. Color coding old SSA values in orange (note that these are regular SSAValues in the uncompacted part, `OldSSAValue` in the compacted part, but not `NewSSAValue` in the uncompacted part, which should print regular).
2. SSA value use accounting. Right not we hide the LHS for compacted nodes that have uses in the uncompacted section.
3. The red separation line should not have a line number.

See this screenshot:
Screen Shot 2022-06-06 at 10 18 28 PM

There are NewSSAValue uses for 8 and 14, but they're not being shown as used in the compacted section.

**Comment by Keno** [comment]
Another corner case bug. We print a pending note that was already compacted twice:
Screen Shot 2022-06-06 at 10 43 27 PM

**Comment by Keno** [comment]
Actually, I'm just wrong about what `NewSSAValue` means in uncompacted IR, but it should probably mean what I want it to, so I'll change that separately.

**Comment by Keno** [comment]
Ok, I've changed what NewSSAValue means in uncompacted IR, so we should appropriately adjust the printing here.

**Comment by aviatesk** [comment]
> I think there is a couple further thing that should be done here:
> 
> 1. Color coding old SSA values in orange (note that these are regular SSAValues in the uncompacted part, `OldSSAValue` in the compacted part, but not `NewSSAValue` in the uncompacted part, which should print regular).
> 2. SSA value use accounting. Right not we hide the LHS for compacted nodes that have uses in the uncompacted section.
> 3. The red separation line should not have a line number.

I think I addressed the items 2&3. As for `NewSSAValue`/`OldSSAValue` stuff, I would like to leave their handling as a future task.

**Pull request merged**